runtime/binding-grpc/src/main/java/io/aklivity/zilla/runtime/binding/grpc/internal/config/GrpcBindingConfig.java

```diff
@@ -107,13 +127,17 @@ public GrpcMethodResult resolveMethod(
             final CharSequence serviceName = serviceNameHeader != null ? serviceNameHeader : matcher.group(SERVICE_NAME);
             final String methodName = matcher.group(METHOD);
 
-            final GrpcMethodConfig method = options.protobufs.stream()
-                .map(p -> p.services.stream().filter(s -> s.service.equals(serviceName)).findFirst().orElse(null))
-                .filter(Objects::nonNull)
-                .map(s -> s.methods.stream().filter(m -> m.method.equals(methodName)).findFirst().orElse(null))
-                .filter(Objects::nonNull)
-                .findFirst()
-                .orElse(null);
+            for (GrpcCatalogSchema catalog : catalogs)
+            {
+                catalog.resolveProtobuf();
+            }
```

Comment:
Lets change `resolveProtobuf()` to return `GrpcProtobufConfig`, then in the `resolveMethod(catalogs)` method, use

```
  .map(GrpcCatalogSchema::resolveProtobuf)
```
instead of 
```
  .map(c -> c.protobuf)
```

So no need for this loop then.

Reply:
Yeah that's what I was thinking thanks good suggestion